README.md

```diff
@@ -1,46 +1,103 @@
 
 In order to make use of the package, you will need to have the following installed:
  - [.NET Core](https://docs.microsoft.com/en-gb/dotnet/core/install/)
  - [Golang](https://golang.org/doc/install)
 
-## Create and use the package
+## Usage
 
-First, you have to make the NuGet Package and use it in your project. To do so, Here are the steps:
-1. execute `make nupkg` from the Makefile
-2. Go to your .NET project and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
-3. Since our library uses Json.Net, you also need to add this dependency to your project. To do so, run the following command: `dotnet add package Newtonsoft.Json`.
+To install SecretHub package from NuGet Gallery, run the following command in your project's direcotry: 
+```bash
+dotnet add package SecretHub
+```
+or you can go to the project's `.csproj` file and add the following line:
+```xml
+<PackageReference Include="SecretHub" Version="0.1.0" />
+```
 
-## How to call library functions
-
-- Read
-`SecretHub.SecretVersion secret = SecretHub.Client.Read("path_to_secret");`
-
-- ReadString
-`string secret = SecretHub.Client.ReadString("path_to_secret");`
-
-- Exists
-`bool isSecret = SecretHub.Client.Exists("path_to_secret");`
-
-- Write
-`SecretHub.Client.Write("path_to_secret", "secret_value");`
+## Building from source 
+1. Execute `make nupkg` from the Makefile
+2. Go to your .NET project direcotry and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
 
-- Remove
-`SecretHub.Client.Remove("path_to_secret");`
-
-- Resolve
-`string secret = SecretHub.Client.Resolve("reference");`
-
-- ResolveEnv
-`Doctionary<string, string> resolvedEnv = SecretHub.Client.ResolveEnv(dictionaryToResolve);`
-
-## Example
-
-An example project is found in `azure-example`. There you can see the library in action.
+## How to call library functions
+Before doing any calls to the library, you need to create you SecretHub client. This is done in the following way:
+```csharp
+var client = new SecretHub.Client();
+``` 
+or 
+```csharp
+SecretHub.Client client = new SecretHub.Client();
+```
+
+After you have your client, you can perform on of the following functions:
+
+### `Read(string path)`
+Retrieve a secret, including all its metadata:
+```csharp
+SecretHub.SecretVersion secret = client.Read("path/to/secret");
+Console.WriteLine("The secret value is " + secret.Data);
+```
+`SecretHub.SecretVersion` object represents a version of a secret with sensitive data.
+
+### `ReadString(string path)`
+Retrieve a secret as a string:
+ ```csharp
+ string secret = client.Read("path/to/secret");
+ Console.WriteLine("The secret value is " + secret);
+ ```
+
+### `Exists(string path)`
+Check if a secret exists at `path`:
+```csharp
+bool isSecret = client.Exists("path/to/secret");
+if (isSecret) 
+{
+    Console.WriteLine("The secret exists.");
+} 
+else 
+{
+    Console.WriteLine("The secret does not exists.");
+}
```

Comment:
I think most readers don't need these last lines to understand what the `Exists()` function does. Less = more :grimacing: 
```suggestion
```

Reply:
Sound good. removed the lines

Reply:
You also removed the first condition. That is fine by me, but then the name of the used variable should be very explanatory. So I'd either:
- Add the first half of the `if` again (with "The secret exists")
or
- Rename the variable to something like `secretExists`